tests/std/tests/P2321R2_views_zip_transform/test.cpp

```diff
@@ -0,0 +1,864 @@
+                STATIC_ASSERT(noexcept(zipped_transformed_range.size()) == noexcept(declval<InnerView&>().size()));
+            }
+
+            STATIC_ASSERT(CanMemberSize<const ZipTransformType> == ranges::sized_range<const InnerView>);
+            if constexpr (CanMemberSize<const ZipTransformType>) {
+                using expected_size_type                            = decltype(declval<const InnerView&>().size());
+                same_as<expected_size_type> auto zip_transform_size = as_const(zipped_transformed_range).size();
+
+                assert(zip_transform_size == ranges::size(transformed_elements));
+                STATIC_ASSERT(
+                    noexcept(zipped_transformed_range.size()) == noexcept(declval<const InnerView&>().size()));
+            }
+
+            const bool is_empty = ranges::empty(transformed_elements);
+
+            // We don't want an empty results range, since we still need to do additional testing.
+            assert(!is_empty);
+
+            // Validate view_interface::empty() and view_interface::operator bool
+            //
+            // From here on out, we'll be re-using concepts which we already verified to reduce
+            // redundancy.
+            STATIC_ASSERT(CanMemberEmpty<ZipTransformType>
+                          == (ranges::sized_range<ZipTransformType> || ranges::forward_range<ZipTransformType>) );
+            if constexpr (CanMemberEmpty<ZipTransformType>) {
+                assert(zipped_transformed_range.empty() == is_empty);
+            }
+
+            STATIC_ASSERT(
+                CanMemberEmpty<const ZipTransformType>
+                == (ranges::sized_range<const ZipTransformType> || ranges::forward_range<const ZipTransformType>) );
+            if constexpr (CanMemberEmpty<const ZipTransformType>) {
+                assert(as_const(zipped_transformed_range).empty() == is_empty);
+            }
+
+            STATIC_ASSERT(CanBool<ZipTransformType> == CanMemberEmpty<ZipTransformType>);
+            if constexpr (CanBool<ZipTransformType>) {
+                assert(static_cast<bool>(zipped_transformed_range) != is_empty);
+            }
+
+            STATIC_ASSERT(CanBool<const ZipTransformType> == CanMemberEmpty<const ZipTransformType>);
+            if constexpr (CanBool<const ZipTransformType>) {
+                assert(static_cast<bool>(as_const(zipped_transformed_range)) != is_empty);
+            }
+
+            // Validate view_interface::cbegin() and view_interface::cend()
+            STATIC_ASSERT(CanMemberCBegin<ZipTransformType>);
+            STATIC_ASSERT(CanMemberCEnd<ZipTransformType>);
+            {
+                STATIC_ASSERT(is_same_v<decltype(zipped_transformed_range.cbegin()),
+                    const_iterator<ranges::iterator_t<ZipTransformType>>>);
+                STATIC_ASSERT(is_same_v<decltype(zipped_transformed_range.cend()),
+                    const_sentinel<ranges::sentinel_t<ZipTransformType>>>);

```

Comment:
Same thing with sentinel: `const_sentinel<ranges::sentinel_t<X>>` to `ranges::const_sentinel_t<X>`.
```suggestion
                    ranges::const_sentinel_t<ZipTransformType>>);
```